runtime/nvqir/cutensornet/simulator_cutensornet.cpp

```diff
@@ -0,0 +1,373 @@
+
+  return counts;
+}
+
+bool SimulatorTensorNetBase::canHandleObserve() { return true; }
+
+static nvqir::CutensornetExecutor *getPluginInstance() {
+  using GetPluginFunction = nvqir::CutensornetExecutor *(*)();
+  auto handle = dlopen(NULL, RTLD_LAZY);
+  GetPluginFunction fcn =
+      (GetPluginFunction)(intptr_t)dlsym(handle, "getCutnExecutor");
+  if (!fcn) {
+    cudaq::info("Externally provided cutensornet plugin not found.");
+    return nullptr;
+  }
+
+  cudaq::info("Successfully loaded the cutensornet plugin.");
+  return fcn();
+}
+/// @brief Evaluate the expectation value of a given observable
+cudaq::ExecutionResult
+SimulatorTensorNetBase::observe(const cudaq::spin_op &ham) {
+  LOG_API_TIME();
+  prepareQubitTensorState();
+  auto *cutnExtension = getPluginInstance();
+  if (cutnExtension) {
+    const auto [terms, coeffs] = ham.get_raw_data();
+    const auto termExpVals =
+        cutnExtension->computeExpVals(m_cutnHandle, m_state->getInternalState(),
+                                      m_state->getNumQubits(), terms);
+    std::complex<double> expVal = 0.0;
+    for (std::size_t i = 0; i < terms.size(); ++i) {
+      expVal += (coeffs[i] * termExpVals[i]);
+    }
+    assert(std::abs(expVal.imag()) < 1e-3);
+    return cudaq::ExecutionResult({}, expVal.real());
+  } else {
+    TensorNetworkSpinOp spinOp(ham, m_cutnHandle);
+    std::complex<double> expVal =
+        m_state->computeExpVal(spinOp.getNetworkOperator());
+    expVal += spinOp.getIdentityTermOffset();
+    assert(std::abs(expVal.imag()) < 1e-3);
+    return cudaq::ExecutionResult({}, expVal.real());
+  }
+}
+
+/// @brief Return the state vector data
+cudaq::State SimulatorTensorNetBase::getStateData() {
+  LOG_API_TIME();
+  if (m_state->getNumQubits() > 64)
+    throw std::runtime_error("State vector data is too large.");
+  // Handle empty state (e.g., no qubit allocation)
+  if (!m_state)
+    return cudaq::State{{0}, {}};
+
+  const std::size_t svDim = 1u << m_state->getNumQubits();
+  return cudaq::State{{svDim}, m_state->getStateVector()};
+}
+
+nvqir::CircuitSimulator *SimulatorTensorNetBase::clone() { return nullptr; }
+void SimulatorTensorNetBase::addQubitsToState(std::size_t count) {
+  LOG_API_TIME();
+  if (!m_state)
+    m_state = std::make_unique<TensorNetState>(count, m_cutnHandle);
+  else if (gateQueue.empty())
+    m_state = std::make_unique<TensorNetState>(m_state->getNumQubits() + count,
+                                               m_cutnHandle);
+  else
+    throw std::runtime_error("Expand qubit register is not supported!");
+}
+void SimulatorTensorNetBase::addQubitToState() {
+  LOG_API_TIME();
+  if (!m_state)
+    m_state = std::make_unique<TensorNetState>(1, m_cutnHandle);
+  else if (gateQueue.empty())
+    m_state = std::make_unique<TensorNetState>(m_state->getNumQubits() + 1,
+                                               m_cutnHandle);
+  else
+    throw std::runtime_error("Expand qubit register is not supported!");
```

Comment:
Can this be simplified?
```suggestion
  addQubitsToState(1);
```